### PR TITLE
panic on negative amount transactionlog, add addlog on receipt

### PR DESCRIPTION
**Pull request by yutongp**

### CODE REVIEW

**Comment by yutongp** [comment]
fix #2351

File: action/receipt.go [hunk 1]
```diff
@@ -119,6 +119,18 @@ func (receipt *Receipt) Hash() hash.Hash256 {
 	return hash.Hash256b(data)
 }
 
+// AddLogs add log to receipt and filter out nil log.
+func (receipt *Receipt) AddLogs(logs ...*Log) {
```
Comment (dustinxie) [inline action/receipt.go:123]: directly use []*Log? so no need to do logs... outside

Comment (yutongp) [reply action/receipt.go:123]: i like this. this is more flexible to add single log

File: action/receipt.go [hunk 2]
```diff
@@ -119,6 +119,18 @@ func (receipt *Receipt) Hash() hash.Hash256 {
...
+	if receipt.Logs == nil {
+		receipt.Logs = make([]*Log, 0, len(logs))
+	}
```
Comment (CoderZhi) [inline action/receipt.go:126]: append works on nil slices, we can delete these three lines

**Pull request merged**